### DISCUSSION TITLE
Metamask not displaying NFTs properly

### DISCUSSION THREAD

**Discussion by luiscaceresd** [Cyfrin/foundry-full-course-cu#1435] [General]
Hi, there learners,

I'm encountering an issue. My NFT is showing up on opensea but not in metamask

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/74666044/a6877fe0-021f-4419-83c3-9045819219f8)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/74666044/5ca39e24-1d86-402a-bb78-e6ce68d162ea)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/74666044/8022378f-42f5-4c78-89fe-3f848d87ad49)

I see that more people are having this issue, could it be a metamask bug? I made sure to enable the show NFT media option on metamask and here's my NFT contract

https://sepolia.etherscan.io/address/0xc04ca966e0d160bfb9ad4acf01435552a17fd342

I appreciate ur help!

**Comment by note-xin** [comment]
for most people 
- delete the imported NFT.
- on meta mask goto **Settings** >> **Advanced** >> `clear activity tab data`.
- in the search bar type `NFT` then enabled `Display NFT Media` and `Auto Detect NFT`.
- import it again.

worked like a charm... 
**(╥_╥)** not for me

**Reply by smahdi-dev** [reply]
@note-xin this worked for me bro

**Reply by p-heisty** [reply]
Thank you! I was testing my local Anvil and Tried Brave, Firefox, and Chrome Metamask. All a default blank image for the NFT.   I created 2x Anvil wallets in Metamask and could transfer the NFT. Funning thing, you could see the image when transfering the NFT back and forth. just not in the wallet...  @note-xin your solution worked perfectly. Thanks again!

**Reply by heeween** [reply]
Thank you so much. I was deploy my NFT token on Sepolia . The picture can't be displayed really waster me too much time to solve. Next time i will check this discussion place when i run into a problem.

**Reply by vishalsingh2972** [reply]
> for most people
> 
> * delete the imported NFT.
> * on meta mask goto **Settings** >> **Advanced** >> `clear activity tab data`.
> * in the search bar type `NFT` then enabled `Display NFT Media` and `Auto Detect NFT`.
> * import it again.
> 
> worked like a charm... **(╥_╥)** not for me

tried this but issue still persists, NFT is visible for me in opensea with image, but nft is visible in metamask but no image is showing in my metamask wallet for the nft.

**Reply by EngrPips** [reply]
Honestly, I would not sweat over this as far as I can see my NFT on opensea. Maybe it is a Metamask Problem.

**Reply by beejaybee** [reply]
Your solution is the best

**Reply by Favourokerri** [reply]
Wow, Thanks, worked like magic

**Reply by EngrPips** [reply]
Awesome.

**Comment by luiscaceresd** [comment]
yeah, didn't work for me either...

**Reply by note-xin** [reply]
how did you define your URI both in the metadata  `.json` and in your interaction script

**Reply by luiscaceresd** [reply]
I'm using Patrick's URI for my json actually so I don't think that's the issue

**Reply by note-xin** [reply]
I went through the discussions a while back... for some users changing `PUG_URI` = `ipfs://bafybeig3...`  with `https://ipfs.io/ipfs/bafybeig` did work.

**Reply by note-xin** [reply]
if you managed to find a solution please help with the same [problem](https://github.com/Cyfrin/foundry-full-course-f23/discussions/1422)

**Comment by Iemberr** [comment]
This question was answered in another discussion, and it worked for me, here it is:
Go to MetaMask → Settings → Security & Privacy.
Enable:
Auto-detect NFTs
Display NFT Media

**Reply by EngrPips** [reply]
Great heads-up.
